python/pyspark/sql/datasource.py

```diff
@@ -45,30 +45,19 @@ class DataSource(ABC):
     """
 
     @final
-    def __init__(
-        self,
-        paths: List[str],
-        userSpecifiedSchema: Optional[StructType],
```

Comment:
why do we also remove user specified schema?

Reply:
This field is actually not used. Both the `reader` and `writer` functions take in the `schema` parameter, and we can pass in the actual schema there.